src/Compilation.zig

```diff
@@ -908,16 +900,17 @@ pub fn create(gpa: *Allocator, options: InitOptions) !*Compilation {
         };
 
         const lto = blk: {
-            if (options.want_lto) |explicit| {
+            if (options.target.isDarwin()) {
+                // TODO ability of zig ld MachO backend to link bitcode files
+                break :blk false;
+            } else if (options.want_lto) |explicit| {
```

Comment:
Hmm, but then, if this is set to `true`, front-end will generate bitcode instead of standard object file leading to weird errors. So you're suggesting checking this and throwing a tailored error instead in the linker?

Reply:
Reverted back to what it was!